src/gui/symbology/qgsgraduatedsymbolrendererwidget.cpp

```diff
@@ -807,16 +862,66 @@ void QgsGraduatedSymbolRendererWidget::classifyGraduated()
   }
 
   QgsGraduatedSymbolRenderer::Mode mode;
-  if ( cboGraduatedMode->currentIndex() == 0 )
+  bool useSymmetricMode = false;
+  bool astride = false;
+
+  int attrNum = mLayer->fields().lookupField( attrName );
+  double minimum = mLayer->minimumValue( attrNum ).toDouble();
+  double maximum = mLayer->maximumValue( attrNum ).toDouble();
+  spinSymmetryPointForOtherMethods->setMinimum( minimum );
+  spinSymmetryPointForOtherMethods->setMaximum( maximum );
+  spinSymmetryPointForOtherMethods->setDecimals( spinPrecision->value() );
+
+  double symmetryPoint = spinSymmetryPointForOtherMethods->value();
+
+  if ( cboGraduatedMode->currentIndex() == 0 ) // EqualInterval
+  {
     mode = QgsGraduatedSymbolRenderer::EqualInterval;
-  else if ( cboGraduatedMode->currentIndex() == 2 )
+    // knowing that spinSymmetryPointForOtherMethods->value() is automatically put at minimum when out of min-max
+    // using "(maximum-minimum)/100)" to avoid direct comparison of doubles
+    if ( spinSymmetryPointForOtherMethods->value() < ( minimum + ( maximum - minimum ) / 100. ) || spinSymmetryPointForOtherMethods->value() > ( maximum - ( maximum - minimum ) / 100. ) )
+      spinSymmetryPointForOtherMethods->setValue( minimum + ( maximum - minimum ) / 2. );
+
+    if ( cbxClassifySymmetric->isChecked() )
+    {
+      useSymmetricMode = true;
+      symmetryPoint = spinSymmetryPointForOtherMethods->value();
+      cbxAstride->isChecked() ? astride = true : astride = false;
+    }
+  }
+  else if ( cboGraduatedMode->currentIndex() == 2 ) // Jenks
+  {
     mode = QgsGraduatedSymbolRenderer::Jenks;
-  else if ( cboGraduatedMode->currentIndex() == 3 )
+  }
+  else if ( cboGraduatedMode->currentIndex() == 3 ) // StdDev
+  {
     mode = QgsGraduatedSymbolRenderer::StdDev;
-  else if ( cboGraduatedMode->currentIndex() == 4 )
+    // knowing that spinSymmetryPointForOtherMethods->value() is automatically put at minimum when out of min-max
+    // using "(maximum-minimum)/100)" to avoid direct comparison of doubles
+    if ( spinSymmetryPointForOtherMethods->value() < ( minimum + ( maximum - minimum ) / 100. ) || spinSymmetryPointForOtherMethods->value() > ( maximum - ( maximum - minimum ) / 100. ) )
+      spinSymmetryPointForOtherMethods->setValue( minimum + ( maximum - minimum ) / 2. );
+
+    if ( cbxClassifySymmetric->isChecked() )
+    {
+      useSymmetricMode = true;
+      symmetryPoint = spinSymmetryPointForOtherMethods->value();
+      cbxAstride->isChecked() ? astride = true : astride = false;
```

Comment:
just go

    astride = cbxAstride->isChecked();

(it's easier to read quickly). This also needs fixing elsewhere in this PR.